components/formats-gpl/src/loci/formats/codec/JPEGXRCodec.java

```diff
@@ -76,7 +81,33 @@
   {
     initialize();
 
-    return service.decompress(buf);
+    byte[] uncompressed = service.decompress(buf);
+
+    if (options.interleaved) {
+      return uncompressed;
+    }
+
+    int bpp = options.bitsPerSample / 8;
+    int pixels = options.width * options.height;
+    int c = uncompressed.length / (pixels * bpp);
+
+    if (c == 1) {
+      return uncompressed;
+    }
+
+    byte[] deinterleaved = new byte[uncompressed.length];
+
+    for (int p=0; p<pixels; p++) {
+      for (int channel=0; channel<c; channel++) {
```

Comment:
Its a very minor nitpick but as someone reading the calculations below my initial thinking was to assume that `channel` would be the total and `c` would be the index, similar to the pixels and bpp loops. Theres nothing wrong with it as is but from readability it might be nicer to swap the variables eg:

`for (int c=0; c<channels; c++) {`

Reply:
Updated in 5405b5b